python/tank/util/yaml_cache.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Implements a caching mechanism to avoid loading the same yaml file multiple times
+unless it's changed on disk.
+"""
+
+import os
+import copy
+import threading
+
+from tank_vendor import yaml
+from ..errors import TankError
+
+class YamlCache(object):
+    """
+    Main yaml cache class
+    """
+    _instance = None
+
+    @staticmethod
+    def instance():
+        """
+        Obtain the single instance of the YamlCache
+        :returns:    The single YamlCache instance
+        """
+        if not YamlCache._instance:
+            YamlCache._instance = YamlCache()
+        return YamlCache._instance
+    
+    def __init__(self):
+        """
+        Construction
+        """
+        self._cache = {}
+        self._lock = threading.Lock()
+            
+    def load(self, path):
```

Comment:
The method name `load` feels little bit misleading because it doesn't always load stuff! :) Is `get` better? Sorry, cannot think of a smashing name right now.... :(